pinot-core/src/main/java/org/apache/pinot/core/data/table/TableResizer.java

```diff
@@ -39,13 +39,13 @@
 /**
  * Helper class for trimming and sorting records in the IndexedTable, based on the order by information
  */
-class IndexedTableResizer {
+class TableResizer {
```

Comment:
will moving this into a separate class outside IndexTable make it harder to optimize later?

Reply:
In this PR, the changes are simply renaming IndexTableResizer to TableResizer.
Pulling out the resize logic to a separate class was done previously. It actually helped in being able to reuse the same logic for both of the IndexedTable implementations. This same class will be used by the Table implementation for distinct. As of now I don't see why it will become harder to optimize. What were you thinking of in particular?

Reply:
Merging this PR, to unblock the Distinct Order By PR.
Will address any specific concerns about this point in separate PRs if needed